pkg/metric/writer_prometheus.go

```diff
@@ -89,6 +97,23 @@ func (w *prometheusWriter) GetPriority() int {
 	return PriorityLow
 }
 
+// shouldFilterMetric drops all metrics which are not passing the configured priority threshold.
+// For cloudwatch (since it does not support summing up over all dimensions for a metric) we are
+// writing total metrics. A total metric is a single timeseries within a metric that has dimensions
+// for other timeseries, such that we can visualize / use the total of everything that we are measuring
+// with that metric. This is not needed on prometheus, as we can just create the total through summing
+// up all our dimensions.
+func (w *prometheusWriter) shouldFilterMetric(datum *Datum) bool {
+	switch {
+	case datum.Priority < w.GetPriority(),
+		datum.Kind == KindTotal:
+		return true
+
+	default:
+		return false
+	}
+}
```

Comment:
Why not use a simple return statement directly?

```suggestion
	return datum.Priority < w.GetPriority() || datum.Kind == KindTotal
}
```